### DISCUSSION TITLE
New image importer for WooCommerce product images

### DISCUSSION THREAD

**Discussion by EricBusch** [datafeedr/wordpress-plugins#10] [Announcements]
Over the past few weeks we've been working on the image importing functionality when importing product images for WooCommerce products.

In the past you may have had products which did not have their images imported for a variety of reasons. We have now addressed many of the most common reasons image importing failed as well as offer a nearly surefire way of getting all of your product images imported.

By default, most product images will be imported.

If you still notice problems with image importing (we noticed this issue with images from Walmart), then you can go one step further and add [Jetpack](https://wordpress.org/plugins/jetpack/) (the free version) to your website. When you do this our plugin will take advantage of the Jetpack's [Photon image service](https://developer.wordpress.com/docs/photon/) which nearly guarantees all valid images are imported for each of your images.

To take advantage of these changes please upgrade the following plugins to their latest versions:

- [Datafeedr API plugin](https://datafeedr.me/dfrapi)
- [Datafeedr Product Sets plugin](https://datafeedr.me/dfrps)
- [Datafeedr WooCommerce Importer plugin](https://datafeedr.me/dfrpswc)

If you prevent images from being imported into your site via custom code, please update your code to the new code [here](https://datafeedrapi.helpscoutdocs.com/article/237-prevent-images-from-being-imported-into-my-website).

Let us know how it works for you!

**Comment by mauspartyhouse** [comment]
Thank you @EricBusch for continuous improvement on the great Datafeedr plugin.

With regards to above improvement; I made following code in the past to import a larger version of the product image.
Since I miss quite some product images on the webshop lately: Should I modify this code?



 * Import a different image field than the default "image" field if a more desired field exists.
 *
 * Some merchants provide a larger version of their images in other fields than the image field. This code allows
 * you to import those large images instead of the smaller ones if the larger ones are available.
 *
 * @param array $meta An array of meta values.
 * @param array $post An array of post data including ID, post_title, post_status, etc...
 * @param array $product An array of product data returned from the Datafeedr API.
 * @param array $set A post array for this Product Set with an array key of postmeta containing all post meta data.
 * @param string $action The action the Product Set is performing. Value are either "insert" or "update".
 *
 * @return array $meta Update $meta array.
 */
 *add_filter( 'dfrpswc_filter_postmeta_array', 'mycode_use_different_image_field', 20, 5 );
 *function mycode_use_different_image_field( $meta, $post, $product, $set, $action ) {
 *
 *	/**
 *	 * An array of fields in the order in which you want to choose them.
 *	 * You can add more fields to this list but do not remove the 'image'
 *	 * field and do not change its position. It should always be last.
 *	 */
 *	$image_fields = array(
 *		'largeimage',       // Use the 'largeimage' field if it exists.
 *		'merchantimageurl', // Use the 'merchantimageurl' field if it exists.
 *		'image',            // DO NOT REMOVE THIS LINE.
 *	);
 *
 *	// Loop through fields in order of usage preference.
 *	foreach ( $image_fields as $field ) {
 *		if ( isset( $product[ $field ] ) ) {
 *			$meta['_dfrps_featured_image_url'] = $product[ $field ];
 *			break;
 *		}
 *	}
 *
 *	return $meta;
 *}
 *

**Comment by EricBusch** [comment]
There's no need to modify that code. It still works as expected.
